modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -357,4 +368,46 @@ Map<String, Long> toMap() {
             return results;
         }
     }
+
+    private static final TimeValue RETRY_STATS_WINDOW = TimeValue.timeValueMinutes(5);
+
+    private static class AwsApiRetryStats {
+        private final LongAdder requests = new LongAdder();
+        private final LongAdder retries = new LongAdder();
+        private final LongAdder throttles = new LongAdder();
+    }
+
+    private volatile AwsApiRetryStats awsApiRetryStats = new AwsApiRetryStats();
+
+    private void maybeLogRetryStats() {
+        var stats = awsApiRetryStats;
+        awsApiRetryStats = new AwsApiRetryStats();
+        long requests = stats.requests.sum();
+        long retries = stats.retries.sum();
+        long throttles = stats.throttles.sum();
+        boolean retried = retries > 0 || throttles > 0;
+        if (retried || logger.isDebugEnabled()) {
+            var message = new ESLogMessage(
+                "Recorded {} requests including {} retries and {} throttles within {}",
+                requests,
+                retries,
+                throttles,
+                RETRY_STATS_WINDOW
+            ).withFields(Map.of("requests", requests, "retries", retries, "throttles", throttles));
+            logger.log(retried ? Level.WARN : Level.DEBUG, message);
+        }
+    }
+
+    private void recordRetryStats(Request<?> request) {
+        var stats = awsApiRetryStats;
+        var info = request.getAWSRequestMetrics().getTimingInfo();
+        stats.requests.add(getCounter(info, AWSRequestMetrics.Field.RequestCount));
+        stats.retries.add(getCounter(info, AWSRequestMetrics.Field.Exception));
```

Comment:
Shouldn't this be `AWSRequestMetrics.Field.RequestCount - 1`? An exception can be thrown and recorded but not retried because reasons like retry limit is reached. Therefore the number of exception is not necessarily the number of retries.

In addition, there is `HttpClientRetryCount` which seems to be the retries at each HTTP request level which might also be relevant?

Reply:
>Shouldn't this be AWSRequestMetrics.Field.RequestCount - 1
If I remember correctly there might be redirects that are also count as additional requests.

We are interested in amount of errors that could happen during the call, I will update the metric to be called exceptions